src/safir/github/_client.py

```diff
@@ -0,0 +1,132 @@
+from __future__ import annotations
+
+from typing import Optional
+
+import gidgethub.apps
+import httpx
+from gidgethub.httpx import GitHubAPI
+
+
+class GitHubAppClientFactory:
+    """Factory for creating GitHub App clients authenticated either as an app
+    or as an installation of that app.
+
+    Parameters
+    ----------
+    id
+        The GitHub App ID.
+    key
+        The GitHub App private key.
+    name
+        The GitHub App name. This identifies the app in the user agent string,
+        and is typically the name of the GitHub repository the app is built
+        from (e.g. ``lsst-sqre/times-square``).
+    http_client
+        The httpx client.
+    """
+
+    def __init__(
+        self, *, id: str, key: str, name: str, http_client: httpx.AsyncClient
+    ) -> None:
+        self.app_id = id
+        self.app_key = key
+        self.app_name = name
+        self._http_client = http_client
+
+    def get_app_jwt(self) -> str:
+        """Create the GitHub App's JWT based on application configuration.
+
+        This token is for authenticating as the GitHub App itself, as opposed
+        to an installation of the app.
+
+        Returns
+        -------
+        str
+            The JWT token.
+        """
+        return gidgethub.apps.get_jwt(
+            app_id=self.app_id, private_key=self.app_key
+        )
+
+    def _create_client(
+        self, *, oauth_token: Optional[str] = None
+    ) -> gidgethub.httpx.GitHubAPI:
```

Comment:
Nit: Here and elsewhere, given that you're importing `GitHubAPI` directly, I would use it directly in the type signature as well, instead of the fully-qualified type name.